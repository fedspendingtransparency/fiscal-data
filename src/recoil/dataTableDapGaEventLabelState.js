import { atom } from 'recoil';

export const dataTableDapGaEventLabelState = atom({
  key: 'dataTableDapGaEventLabelState',
  default: '',
});
