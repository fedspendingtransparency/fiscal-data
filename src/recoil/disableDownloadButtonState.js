import { atom } from 'recoil';

export const disableDownloadButtonState = atom({
  key: 'disableDownloadButtonState',
  default: false,
});
