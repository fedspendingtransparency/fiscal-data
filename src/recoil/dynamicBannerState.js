import { atom, selector } from "recoil";
import { apiPrefix, basicFetch } from "../utils/api-utils";

export const dynamicBannerLastCachedState = atom({
  key: 'dynamicBannerLastCachedState',
  default: 0,
});

export const dynamicBannerState = selector({
  key: 'dynamicBannerState',
  get: async () => {
    const announcementsEndpoint = 'v1/reference/fiscal_data/announcements';
    const response = await basicFetch(`${apiPrefix}${announcementsEndpoint}`);
    const timeStamp = Date.now();
    return {
      payload: response.data,
      timeStamp: timeStamp,
    };
  },
});
