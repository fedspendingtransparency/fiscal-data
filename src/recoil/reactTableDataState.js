import { atom } from 'recoil';

export const reactTablePageState = atom({
  key: 'reactTablePage',
  default: 0,
});

export const reactTableDePaginatedDataState = atom({
  key: 'reactTableDePaginatedDataState',
  default: null,
});
