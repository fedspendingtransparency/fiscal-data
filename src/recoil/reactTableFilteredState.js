import { atom } from 'recoil';

export const reactTableFilteredDateRangeState = atom({
  key: 'reactTableFilteredDateRangeState',
  default: null,
});

export const reactTableSortingState = atom({
  key: 'reactTableSortingState',
  default: [],
});
