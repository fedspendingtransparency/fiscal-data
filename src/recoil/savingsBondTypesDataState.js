import { selector, atom } from 'recoil';
import { apiPrefix, basicFetch } from '../utils/api-utils';

export const savingsBondTypesLastCachedState = atom({
  key: 'savingsBondTypesLastCachedState ',
  default: 0,
});

export const savingsBondTypesData = selector({
  key: 'savingsBondTypesData',
  get: async ({ get }) => {
    const numOfSavingsBonds = 'v1/accounting/od/securities_sales?filter=security_type_desc:eq:Savings%20Bond';
    const response = await basicFetch(`${apiPrefix}${numOfSavingsBonds}&page[size]=1`).then(metaRes => {
      if (metaRes.meta && typeof metaRes.meta['total-pages'] !== 'undefined') {
        const pageSize = metaRes.meta['total-pages'];
        return basicFetch(`${apiPrefix}${numOfSavingsBonds}&page[size]=${pageSize}`);
      }
    });
    const timeStamp = Date.now();
    return {
      payload: response.data,
      timeStamp: timeStamp,
    };
  },
});
