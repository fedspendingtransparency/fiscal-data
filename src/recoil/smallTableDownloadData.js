import { atom } from 'recoil';

export const smallTableDownloadDataCSV = atom({
  key: 'smallTableDownloadDataCSV',
  default: [],
});

export const smallTableDownloadDataJSON = atom({
  key: 'smallTableDownloadDataJSON',
  default: '',
});

export const smallTableDownloadDataXML = atom({
  key: 'smallTableDownloadDataXML',
  default: {},
});

export const tableRowLengthState = atom({
  key: 'tableRowLengthState',
  default: null,
});
