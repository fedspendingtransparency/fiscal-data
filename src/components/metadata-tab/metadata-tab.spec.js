import React from 'react';
import renderer from 'react-test-renderer';
import MetadataTab from './metadata-tab';
import DtgTable from '../dtg-table/dtg-table';
import { RecoilRoot } from 'recoil';

describe('MetadataTab', () => {
  const mockConfig = {
    name: 'metadata test',
    summaryText: 'long description of the metadata test',
    tagLine: 'shorter description',
    updateFrequency: 'frequently',
    techSpecs: {
      earliestDate: '12/12/12',
      latestDate: '1/1/20',
    },
    relatedTopics: ['topic1', 'topic2'],
    slug: '/detailed-url-info/',
  };

  let component = renderer.create();
  let instance;
  beforeAll(() => {
    renderer.act(() => {
      component = renderer.create(
        <RecoilRoot>
          <MetadataTab config={mockConfig} />
        </RecoilRoot>
      );
    });
    instance = component.root;
  });

  it('should pass along its data array to the dtgTable component', () => {
    const tableData = instance.findByType(DtgTable).props.tableProps.data;
    expect(tableData).toMatchSnapshot();
  });

  it('sets aria-label to dataset name metadata', () => {
    expect(instance.findByType('table').props['aria-label']).toBe(`${mockConfig.name} metadata`);
  });
});
