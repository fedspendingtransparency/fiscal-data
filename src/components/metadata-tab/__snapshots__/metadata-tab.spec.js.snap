// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetadataTab should pass along its data array to the dtgTable component 1`] = `
[
  {
    "definition": "metadata test",
    "name": "Title",
  },
  {
    "definition": "long description of the metadata test",
    "name": "Description (Long)",
  },
  {
    "definition": "shorter description",
    "name": "Description (Short)",
  },
  {
    "definition": undefined,
    "name": "Update Frequency",
  },
  {
    "definition": "12/12/12 - 1/1/20",
    "name": "Date Range",
  },
  {
    "definition": "topic1, topic2",
    "name": "Topics",
  },
  {
    "definition": undefined,
    "name": "Publisher",
  },
]
`;
